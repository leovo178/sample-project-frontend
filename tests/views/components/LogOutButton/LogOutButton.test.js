import React from 'react'
import { cleanup, fireEvent, render } from '@testing-library/react'

import { LogOutButton } from '@views_components'

describe('<LogOutButton />', () => {
	const mockProps = {
		handleOnLogOut: jest.fn(),
	}

	afterEach(() => {
		cleanup()
	})

	it('should match snapshot', () => {
		const { container } = render(<LogOutButton {...mockProps} />)

		expect(container).toMatchSnapshot()
	})

	it('should call handleOnLogOut correctly when click', () => {
		const { getByTestId } = render(<LogOutButton {...mockProps} />)
		const logoutButton = 'logoutbutton'

		fireEvent.click(getByTestId(logoutButton))

		expect(mockProps.handleOnLogOut).toHaveBeenCalled()
	})
})
