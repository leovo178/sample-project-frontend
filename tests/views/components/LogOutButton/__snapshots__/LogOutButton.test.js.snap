// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogOutButton /> should match snapshot 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-root-1 MuiButton-textPrimary"
    data-testid="logoutbutton"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Logout
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
